### ISSUE TITLE
Core performance tests improvements (2025 edition)

### ISSUE THREAD

**Issue by swissspidy** [WordPress/performance#1735]
We talked about potential improvements to the core performance test suite in the past, but haven't worked on it much so far.

After we saw some slight performance regressions after 6.7 not only [in lab data](https://make.wordpress.org/core/2024/12/10/wordpress-6-7-performance-improvements/) but also in the field, now is a good time to at least work on some quick wins.

> [!NOTE]  
> This is split out from #1093, which is a larger ticket with long discussions about multiple aspects of performance testing.

Q1:

* [x] Update baseline in tests to 6.7
* [x] Test TT5 theme
* [x] Contribute improvements to [the codevitals repo](https://github.com/youknowriad/codevitals) to make it less overwhelming when there are more stats (e.g. a checkbox to hide/show certain fields)
* [x] Test single post URLs as well.

Q2:

* [ ] Raise awareness of the dashboard among the team by adding to the handbook and any checklists
* [ ] Emulate broadband network conditions in core performance tests
	* Requires https://github.com/WordPress/gutenberg/pull/69865

Q3:

* [ ] Improve dummy content to be more realistic and document it

Stretch goals:

* [ ] Build some alerting if nothing gets reported or if there is a drastic change in metrics

**Comment by joemcgill** [comment]
> Test TT5 theme

There is https://core.trac.wordpress.org/ticket/62148 open to track this, along with [this PR](https://github.com/WordPress/wordpress-develop/pull/7688/) that needs to be reviewed.

**Comment by felixarntz** [comment]
Thanks @swissspidy for the great summary!

A few additional thoughts:
* Ideas for enhancing the Code Vitals dashboard UI to make it easier to overview and use:
    * Select <=4 key metrics to highlight, with all other ones being shown in a less prominent way (e.g. "advanced" UI, or simply in a different place, or smaller). There are way too many metrics right now all presented equally, but some of them are far less important than others.
        * My first intuition would be to highlight "LCP - TTFB" (client-side performance) and _either_ "TTFB" _or_ "Server-Timing total" (server-side performance).
        * Both of these metrics could be highlighted for the most recent block theme and the most recent classic theme, which would give us a total of 4 key metrics.
    * Instead of the "block theme" and "classic theme" labeling, use the actual theme names (or slugs).
        * Related: Consider defining a "policy" on which themes get included, e.g. the 3 most recent default block themes and default classic themes (the latter would probably forever be 2021, 2020, and 2019).
    * Separate metrics for different "products" (e.g. the different themes) in different tabs.
* I think we should remove the point about the TTFB regression, as that's an actual performance issue possibly related to 6.7 - I think that would be best in its own issue decoupled from this, as it has nothing to do really with improving the metric collection and dashboard.

**Comment by swissspidy** [comment]
> Instead of the "block theme" and "classic theme" labeling, use the actual theme names (or slugs)

As discussed, we can do this while preserving "backward compatibility" with the data, i.e. we can keep the existing `home-classic-theme` and `home-block-theme` tests for the current themes we're testing, and then use the actual theme slugs for the new ones.

**Comment by swissspidy** [comment]
> Improve dummy content to be more realistic and document it

The homepages are currently suboptimal with the theme unit test data. Either it's a full list of the 10 latest posts (which is gigantic) or it's the "Front page" page, which just contains 1 paragraph and nothing else.

> Test single post URLs as well.

In the theme unit test data there is a "Block: Image" post with multiple images in them that I think is a reasonable start:

![Image](https://github.com/user-attachments/assets/e3457704-0d67-48b9-8c3b-9f5a7d2735ed)

**Comment by felixarntz** [comment]
@swissspidy 
> > Test single post URLs as well.
> 
> In the theme unit test data there is a "Block: Image" post with multiple images in them that I think is a reasonable start:

Agreed. Though I'd love for this post to also get a featured image, as that makes a big impact on performance for single post URLs in most themes - or alternatively, use that post but with the image as the very first block. Currently, I think the image is too far down the page to be in the initial viewport for most mobile devices.

**Comment by swissspidy** [comment]
> Currently, I think the image is too far down the page to be in the initial viewport for most mobile devices.

FWIW we don't actually test mobile at all. All tests run using [Desktop Chrome](https://github.com/microsoft/playwright/blob/7ce1a540bcd979096dc4ac3851c675407ede6f5a/packages/playwright-core/src/server/deviceDescriptorsSource.json#L1624-L1638) with a 1280x720 viewport. (Though there the image is not in the initial viewport either)

**Comment by swissspidy** [comment]
I updated the issue description to include the main goals for this year.

Network emulation is something that has just come up recently (cc @westonruter), so I opened https://github.com/WordPress/gutenberg/pull/69865 with the goal of adding this to the core performance tests.
